src/go/rpk/pkg/cli/topic/analyze.go

```diff
@@ -0,0 +1,743 @@
+			secBatchRate: cfg.PrintTopicSummary,
+			secBatchSize: cfg.PrintTopicSummary,
+		})...,
+	)
+
+	sections.Add(secSummary, func() {
+		tw := out.NewTabWriter()
+		tw.PrintColumn("topics", gs.Topics)
+		tw.PrintColumn("partitions", gs.Partitions)
+		tw.PrintColumn("total batch rate (1/s)", gs.TotalBatchesPerSecond)
+		tw.PrintColumn("p50 batch size (bytes)", gs.BatchSizes[0].value)
+		tw.Flush()
+	})
+
+	sections.Add(secBatchRate, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchesPerSecond {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+
+	sections.Add(secBatchSize, func() {
+		tw := out.NewTable(append([]string{"TOPIC"}, percentileNames...)...)
+		defer tw.Flush()
+		for _, ts := range ts {
+			vals := []interface{}{}
+			for _, pv := range ts.BatchSizes {
+				vals = append(vals, pv.value)
+			}
+
+			tw.Print(append([]interface{}{ts.Topic}, vals...)...)
+		}
+	})
+	return nil
+}
+
+type kvTPInfo struct {
+	TopicPartition topicPartition      `json:"topicPartition" yaml:"topic_partition"`
+	Info           *topicPartitionInfo `json:"info" yaml:"info"`
+}
+
+type rawResults struct {
+	TopicPartitionInfo []kvTPInfo `json:"topicPartitionInfo" yaml:"topic_partition_info"`
+	TimeRange          timeRange  `json:"timeRange" yaml:"time_range"`
+}
+
+func (cr collectResult) printRawResults(f config.OutFormatter, w io.Writer) (bool, error) {
+	// Json doesn't seem to like serializing the map types. So convert it to a slice here.
+	kvs := []kvTPInfo{}
+	for k, v := range cr.TopicPartitionInfo {
+		kvs = append(kvs, kvTPInfo{k, v})
+	}
+
+	res := rawResults{kvs, cr.TimeRange}
+	if isText, _, t, err := f.Format(res); !isText {
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Fprintln(w, t)
+		return true, nil
+	}
+	return false, nil
+}
+
+const helpAnalyze = `Analyze topics.
+
+This command consumes records from the specified topics in order to determine
+characteristics of the topic like batch rate and batch size.
+
+Using the --format flag with either JSON or YAML prints all of the metadata collected.
+
+TOPICS
+
+Topics can either be listed individually or by regex via the --regex flag (-r).
+
+For example,
+
+    analyze foo bar            # describe topics foo and bar
+    analyze -r '^f.*' '.*r$'   # describe any topic starting with f and any topics ending in r
+    analyze -r '*'             # describe all topics
+    analyze -r .               # describe any one-character topics
+
+TIME RANGE
+
+The --time-range flag allows for specifying what time range to consume from.
+The time range should be of the following format:
+
+    t1:t2    consume from timestamp t1 until timestamp t2
+
+There are a few options for timestamps, with each option being evaluated
+until one succeeds:
+
+    13 digits             parsed as a unix millisecond
```

Comment:
```suggestion
    13 digits             parsed as a Unix millisecond
```
I think this would technically have to be UNIX per the official trademark but it doesn't seem like we've standardized docs around that yet, so Unix should be ok for now

Reply:
Alright will change